velox/functions/prestosql/tests/InPredicateTest.cpp

```diff
@@ -317,3 +320,25 @@ TEST_F(InPredicateTest, varbinary) {
   auto result = evaluate<SimpleVector<bool>>(predicate, input);
   assertEqualVectors(makeConstant(true, input->size()), result);
 }
+
+TEST_F(InPredicateTest, reusableResult) {
+  auto a1 = makeNullableFlatVector<int32_t>({0, 1, std::nullopt, 3, 4, 5, 6});
+  auto b1 = makeFlatVector<int64_t>({0, 5, 10, 15, 20, 25, 30});
+  auto data1 = makeRowVector({"a", "b"}, {a1, b1});
+
+  auto a2 = makeNullableFlatVector<int32_t>({0, 1, 2, 3, 4, 5});
+  auto b2 = makeFlatVector<int64_t>({100, 105, 110, 115, 120, 125});
+  auto data2 = makeRowVector({"a", "b"}, {a2, b2});
+
+  std::shared_ptr<memory::MemoryPool> pool_{memory::addDefaultLeafMemoryPool()};
+  auto plan = PlanBuilder(pool_.get())
```

Comment:
There is not need to run full query plan to reproduce this issue. Let's create a version of 'evaluate' that takes a 'result' vector and use that instead.